src/scripts/main.js

```diff
@@ -1,3 +1,233 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tbody = table.querySelector('tbody');
+const rows = [...tbody.querySelectorAll('tr')];
+const form = document.createElement('form');
+let inputActive = false;
+let textValue = '';
+
+form.classList.add('new-employee-form');
+
+form.insertAdjacentHTML('afterbegin', `
+  <label>Name: <input name="name" type="text" data-qa="name" required></label>
+  <label>Position: 
+    <input
+      name="position"
+      type="text"
+      data-qa="position"
+      required
+    >
+  </label>
+  <label>Office: 
+    <select name="office" data-qa="office" required>
+      <option>Tokyo</option>
+      <option>Singapore</option>
+      <option>London</option>
+      <option>New York</option>
+      <option>Edinburgh</option>
+      <option>San Francisco</option>
+    </select>
+  </label>
+  <label>Age: 
+    <input name="age" type="number" data-qa="age" required>
+  </label>
+  <label>Salary: 
+    <input name="salary" type="number" data-qa="salary" required>
+  </label>
+  <button>Save to table</button>
+`);
+
+table.parentNode.insertBefore(form, table.nextSibling);
+
+table.querySelectorAll('thead th').forEach((th, index) => {
+  th.addEventListener('click', (ev) => {
+    sortColumn(index, ev);
+  });
+});
+
+function normalizeValue(el, index, isNumeric) {
+  const value = el.cells[index].textContent.replace(/[$,]/g, '');
+
+  return isNumeric ? +value : value;
+}
+
+function sortColumn(index, ev) {
+  ev.target.setAttribute('data-direction',
+    ev.target.getAttribute('data-direction')
+  && ev.target.getAttribute('data-direction') === 'asc' ? 'desc' : 'asc');
```

Comment:
it's hard to read so it would be better to create variables and use it in `setAttribute`